pkg/version/version_info.go

```diff
@@ -0,0 +1,68 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type versionInfo struct {
+	major   int
+	minor   int
+	release int
+}
+
+// compareVersionInfo returns:
+// 	0: if a == b
+//  n > 0: if a > b
+//  n < 0: if a < b
+//  0 with error: if a == nil || b == nil
+func compareVersionInfo(a *versionInfo, b *versionInfo) (int, error) {
+	if a == nil || b == nil {
+		return 0, fmt.Errorf("parameter must not be null")
+	}
+
+	// Check major version
+	result := a.major - b.major
+	if result != 0 {
+		return result, nil
+	}
+
+	// Major is equal, check minor
+	result = a.minor - b.minor
+	if result != 0 {
+		return result, nil
+	}
+
+	// Major and minor is equal, check release
+	result = a.release - b.release
+	return result, nil
+}
+
+func extractVersion(versionString string) (*versionInfo, error) {
+	mainVersionRegex := regexp.MustCompile(`[\d]+\.[\d]+\.[\d]+`)
```

Comment:
An example version is `1.203.12.20200914-092723` - I wonder if, for development pipelines, the first three fields remain the same, but the later changes.

Also, `regexp.MustCompile` is to be used as a global variable.